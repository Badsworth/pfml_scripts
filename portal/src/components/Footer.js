import FooterSlim from "@massds/mayflower-react/dist/FooterSlim";
import React from "react";
import SiteLogo from "@massds/mayflower-react/dist/SiteLogo";
import logo from "@massds/mayflower-assets/static/images/logo/stateseal.png";
import routes from "../routes";
import { useTranslation } from "../locales/i18n";

/**
 * Global page footer, displayed at the bottom of every page.
 */
const Footer = () => {
  const { t } = useTranslation();

  const footerSlimProps = {
    title: t("components.footer.title"),
    description: t("components.footer.description"),
    contact: {
      address: t("components.footer.orgAddress"),
      phone: t("components.footer.orgPhoneNumber"),
      online: {
        title: t("components.footer.orgName"),
        href: routes.external.massgov.dfml,
      },
    },
    siteLogo: (
      <SiteLogo
        url={{
          domain: routes.home,
        }}
        image={{
          src: logo,
          alt: t("components.siteLogo.alt"),
          width: 45,
          height: 45,
        }}
        title={`${t("components.header.appTitle")} Homepage`}
      />
    ),
  };

  return <FooterSlim {...footerSlimProps} />;
};

export default Footer;
