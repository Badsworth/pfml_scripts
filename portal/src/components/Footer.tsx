import DFMLLogo from "./DFMLLogo.svg";
import FooterSlim from "@massds/mayflower-react/dist/FooterSlim";
import PFMLLogo from "./PFMLLogo.svg";
import React from "react";
import SiteLogo from "@massds/mayflower-react/dist/SiteLogo";
import routes from "../routes";
import { useTranslation } from "../locales/i18n";

/**
 * Global page footer, displayed at the bottom of every page.
 */
const Footer = () => {
  const { t } = useTranslation();

  const footerSlimProps = {
    title: t("components.footer.title"),
    description: t("components.footer.description"),
    contact: {
      address: t("components.footer.orgAddress"),
      phone: t("components.footer.orgPhoneNumber"),
      online: {
        title: t("components.footer.orgName"),
        href: routes.external.massgov.dfml,
      },
    },
    links: [
      {
        href: routes.external.massgov.privacyPolicy,
        title: t("components.footer.privacyPolicy"),
      },
      {
        href: routes.external.massgov.consentAgreement,
        title: t("components.footer.dataSharingAgreement"),
      },
    ],
    stackedLogo: true,
    siteLogo: (
      <React.Fragment>
        <SiteLogo
          url={{
            domain: routes.external.massgov.dfml,
          }}
          image={{
            src: DFMLLogo,
            alt: t("components.siteLogo.dfmlAlt"),
            width: 165,
            height: 45,
          }}
          title={t("components.footer.logoTitleDFML")}
        />
        <SiteLogo
          url={{
            domain: routes.external.massgov.pfml,
          }}
          image={{
            src: PFMLLogo,
            alt: t("components.siteLogo.pfmlAlt"),
            width: 118,
            height: 45,
          }}
          title={t("components.footer.logoTitlePFML")}
        />
      </React.Fragment>
    ),
  };

  return <FooterSlim {...footerSlimProps} />;
};

export default Footer;
