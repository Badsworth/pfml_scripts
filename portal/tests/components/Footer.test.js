import Footer from "../../src/components/Footer";
import React from "react";
import { render } from "@testing-library/react";

describe("Footer", () => {
  it("renders footer with default settings", () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
