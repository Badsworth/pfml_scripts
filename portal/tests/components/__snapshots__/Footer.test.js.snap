// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer with default settings 1`] = `
<FooterSlim
  contact={
    Object {
      "address": "PO Box 838 Lawrence, MA 01843",
      "online": Object {
        "href": "https://mass.gov/dfml",
        "title": "Department of Family and Medical Leave (DFML)",
      },
      "phone": "(833) 344‑7365",
    }
  }
  description="Paid Family and Medical Leave is a state-offered benefit for anyone who works in Massachusetts and is eligible to take up to 26 weeks of paid leave for medical or family reasons."
  siteLogo={
    <React.Fragment>
      <SiteLogo
        image={
          Object {
            "alt": "Department of Family and Medical Leave logo",
            "height": 45,
            "src": "test-file-stub",
            "width": 165,
          }
        }
        title="Go to PFML homepage"
        url={
          Object {
            "domain": "/",
          }
        }
      />
      <SiteLogo
        image={
          Object {
            "alt": "Paid Family and Medical Leave program logo",
            "height": 45,
            "src": "test-file-stub",
            "width": 118,
          }
        }
        title="Go to PFML homepage"
        url={
          Object {
            "domain": "/",
          }
        }
      />
    </React.Fragment>
  }
  stackedLogo={true}
  title="Paid Family and Medical Leave (PFML)"
/>
`;
